### PR TITLE
Finished porting of "t_pub/sub_delay"

### PR DESCRIPTION
**Pull request by eddie9712**
* Replace structopt to clap in `t_pub/sub_delay.rs`
* Update the usage of zenoh in `t_pub/sub_delay.rs`
* Add payload size checking (payload size should >= 24)
* Cargo fmt

- [x] Compilation passed with command cargo build --bin t_pub(sub)
_delay
- [x] Verified by the following commands: `./t_sub_delay -e tcp/127.0.0.1:7447 -m peer`,
`./t_ping -m peer -p 24 -i 1 -e tcp/127.0.0.1:7447`
- [x] Checked with Cargo clippy

### CODE REVIEW

**Comment by william-wyliang** [comment]
Also invite @YuanYuYuan to review.

File: latency/src/bin/t_pub_delay.rs [hunk 1]
```diff
@@ -91,14 +98,17 @@ async fn main() {
             reliability: Reliability::Reliable,
         };
         let congestion_control = CongestionControl::Block;
-        let key = ResKey::RName("/test/ping".to_string());
+        let key = "/test/ping".into();
```
Comment (YuanYuYuan) [inline latency/src/bin/t_pub_delay.rs:101]: A plain `into` is meaningless. I suggest you can move to where `key` is used.

Comment (eddie9712) [reply latency/src/bin/t_pub_delay.rs:101]: Thanks, I will edit it.

File: latency/src/bin/t_pub_delay.rs [hunk 2]
```diff
@@ -98,14 +100,17 @@ async fn main() {
...
         let info = None;
         let routing_context = None;
         let reply_context = None;
         let attachment = None;
 
         // u64 (8 bytes) for seq num
         // u128 (16 bytes) for system time in nanoseconds
+        if opt.payload < 24 {
```
Comment (william-wyliang) [inline latency/src/bin/t_pub_delay.rs:109]: Good to check here.

File: latency/src/bin/t_sub_delay.rs [hunk 3]
```diff
@@ -60,22 +63,28 @@ impl MyMH {
...
+                            .as_nanos();
+                        let interval = Duration::from_nanos((now_sub - now_pub) as u64);
+
+                        println!("{} bytes: seq={} time={:?}", payload.len(), count, interval);
+                    } else {
+                        panic!("Fail to fill the buffer");
+                    }
+                } else {
+                    panic!("Fail to fill the buffer");
+                }
```
Comment (william-wyliang) [inline latency/src/bin/t_sub_delay.rs:87]: Consider to refactor this segment of code. Checking the success of `read_exact()` of `count` and `now_pub` all before `now_sub`.

**Review by william-wyliang — CHANGES_REQUESTED** [review]
Suggest to do some code refactoring. See the inline comments.

File: latency/src/bin/t_pub_delay.rs [hunk 4]
```diff
@@ -52,16 +52,23 @@ impl TransportEventHandler for MySH {
     }
 }
 
-#[derive(Debug, StructOpt)]
-#[structopt(name = "z_ping")]
+#[derive(Debug, Parser)]
+#[clap(name = "t_pub_delay")]
```
[the remark this answers is not in the document]
Comment (eddie9712) [reply latency/src/bin/t_pub_delay.rs:53]: I think it is a typo, the `name` is the binary name displayed in help messages.

File: latency/src/bin/t_sub_delay.rs (hunk 3 above)
Comment (eddie9712) [reply latency/src/bin/t_sub_delay.rs:86]: Ok, I will try to reduce the indent of this segment.

**Review by william-wyliang — APPROVED** [review]

**Pull request merged**